### ISSUE TITLE
Support for snapshots, or "Save copy of project"

### ISSUE THREAD

**Issue by Tisten** [OurMachinery/themachinery-public#1178]
I.e don't let the editor set the save project as the current project, and keep the dirty state as is.

When connected to a collaboration server, this should be the way to save the project. When doing "Save Project As..." right now the editor will both get updates from the server, and also watch the saved files on disk and suggesting to reload the project if they change.

I can make a Pull Request for this feature later this week.

Side note: I'm using this kind of save for doing automatic snaphots in the headless server, combining it with utilizing hard links towards a "Content Addressable Store" which stores buffers in the same kind of layout as the "collaboration cache".

**Comment by niklas-ourmachinery** [comment]
Sounds good.

Buuut... a save shouldn't normally trigger the "changed files on disk" alert -- that would be super annoying. Instead, the alert is only supposed to be triggered if the file change in some other way ... such as if you do a `git pull` in the project directory.

Admittedly, we don't have a very good algorithm for detecting this, just some heuristics, because AFAICT there is no easy "right" way of doing it... but we may need to adjust those heuristics or do something else if you keep getting those messages.

You could also potentially turn the change detection off if you're not using it.

**Comment by Tisten** [comment]
Well, the "changed files on disk" might have triggered because they changed. When I need to debug the collaboration project I normally do a "Save Project As...", but keep the editor connected as a client. Then I start another editor and load the saved data and try to find things out. And sometimes I save in the second editor.

What I'm thinking is that the first editor should still consider only the collaboration server as the project provider, and not both the server and the saved project. I've fixed that in our local version, but anytime I work with bug reporting I try to do that against your master without our changes, and then I get this "import changes from disc" popup again. I don't want to disable it because I will probably forget to enable it again. But I never want the popup when connected to collaboration. But that is a separate issue from the feature suggestion I made above.
